### DISCUSSION TITLE
Is it necessary to enhance the differentiation of different file contents in Vault QA mode?

### DISCUSSION THREAD

**Discussion by wwjCMP** [logancyang/obsidian-copilot#542] [Q&A]
The related segments from different files are now packaged and sent together to the LLM, making it seem unable to distinguish the ownership of the segments.

For example, suppose I have a film/actor database, and I want the LLM to help me recommend specific similar films or actors. Due to the weak structure of the segments, it confuses information that belongs to different actors.

**Comment by logancyang** [comment]
Each segment (chunk) from a note has the note title in it. The LLM can see them. 

If "it seem unable to distinguish the ownership of the segments", it could be a weak chat model having bad understanding. GPT-4 level models shouldn't have a problem.

**Comment by wwjCMP** [comment]
Understanding is one aspect, but I think the most important thing is that file names can be explicitly marked with text, not just titles + \n\n, because titles are likely to be confused with in-text bi-directional links. If Copilot also pays attention to local LLMs, this improvement would be meaningful.
